src/nvtxw.rs

```diff
@@ -0,0 +1,385 @@
+            },
+            nvtxw::PayloadSchemaEntry {
+                flags: nvtxw::NVTX_PAYLOAD_ENTRY_FLAG_EVENT_MESSAGE
+                    | nvtxw::NVTX_PAYLOAD_ENTRY_FLAG_POINTER,
+                type_: nvtxw::NVTX_PAYLOAD_ENTRY_TYPE_CSTRING,
+                name: c_field_name_3.as_ptr(),
+                description: null(),
+                arrayOrUnionDetail: 0,
+                offset: 0,
+                semantics: null(),
+                reserved: null(),
+            },
+            nvtxw::PayloadSchemaEntry {
+                flags: 0,
+                type_: nvtxw::NVTX_PAYLOAD_ENTRY_TYPE_NVTX_COLOR,
+                name: c_field_name_4.as_ptr(),
+                description: null(),
+                arrayOrUnionDetail: 0,
+                offset: 0,
+                semantics: null(),
+                reserved: null(),
+            },
+        ];
+
+        let schema_attr = nvtxw::PayloadSchemaAttr {
+            fieldMask: nvtxw::NVTX_PAYLOAD_SCHEMA_ATTR_NAME
+                | nvtxw::NVTX_PAYLOAD_SCHEMA_ATTR_TYPE
+                | nvtxw::NVTX_PAYLOAD_SCHEMA_ATTR_FLAGS
+                | nvtxw::NVTX_PAYLOAD_SCHEMA_ATTR_ENTRIES
+                | nvtxw::NVTX_PAYLOAD_SCHEMA_ATTR_NUM_ENTRIES
+                | nvtxw::NVTX_PAYLOAD_SCHEMA_ATTR_STATIC_SIZE
+                | nvtxw::NVTX_PAYLOAD_SCHEMA_ATTR_SCHEMA_ID,
+            name: c_event_name.as_ptr(),
+            type_: nvtxw::NVTX_PAYLOAD_SCHEMA_TYPE_STATIC,
+            flags: nvtxw::NVTX_PAYLOAD_SCHEMA_FLAG_NONE,
+            entries: evt_schema.as_ptr(),
+            numEntries: evt_schema.len(),
+            payloadStaticSize: size_of::<legion_nvtxw_event>(),
+            packAlign: 0,
+            schemaId: LEGION_NVTXW_PAYLOAD_SCHEMA,
+            extension: null_mut(),
+        };
+
+        let mut streams: Vec<Vec<Vec<nvtxw::StreamHandle>>> = Vec::new();
+        for str0 in panels {
+            let mut x0: Vec<Vec<nvtxw::StreamHandle>> = Vec::new();
+            for str1 in str0 {
+                let mut x1: Vec<nvtxw::StreamHandle> = Vec::new();
+                for (stream_name, domain_name) in str1 {
+                    let stream =
+                        nvtxw::stream_begin_simple(interface, session, stream_name, domain_name)
+                            .expect("Failed to create stream");
+
+                    nvtxw::schema_register(interface, session, stream, &schema_attr)
+                        .expect("Failed to register schema");
+
+                    x1.push(stream);
+                }
+
+                x0.push(x1);
+            }
+            streams.push(x0);
+        }
+
+        let zero_time = self.zero_time;
+
+        for (_, i) in entries {
+            let c_name = CString::new(i.title).expect("CString::new failed");
+            let events = [legion_nvtxw_event {
+                time_start: (i.time_start.0 as u64).wrapping_add_signed(zero_time),
+                time_stop: (i.time_stop.0 as u64).wrapping_add_signed(zero_time),
```

Comment:
Are you sure you want a wrapping add here? While it seems unlikely, if (somehow) we actually hit this case, the resulting times will presumably be totally bogus.... It seems better to me to use checked math:

```
i.time_start.0.try_into().unwrap().checked_add(zero_time)
```

Not pretty but it ensures you throw an error if something unexpected happens.

Reply:
Done